src/IB/IBFEMethod.cpp

```diff
@@ -1278,1241 +1107,739 @@ void
-                // we also need workload
-                if (!scratch_level->checkAllocated(d_lagrangian_workload_current_idx))
-                    scratch_level->allocatePatchData(d_lagrangian_workload_current_idx);
-            }
+/////////////////////////////// PROTECTED ////////////////////////////////////
 
-            HierarchyCellDataOpsReal<NDIM, double> hier_cc_data_ops(
-                d_scratch_hierarchy, 0, d_scratch_hierarchy->getFinestLevelNumber());
-            hier_cc_data_ops.setToScalar(d_lagrangian_workload_current_idx, 0.0);
-            addWorkloadEstimate(d_scratch_hierarchy, d_lagrangian_workload_current_idx);
+void
+IBFEMethod::doInitializeFEEquationSystems()
+{
+    const bool from_restart = RestartManager::getManager()->isFromRestart();
 
-            // Use this class' buffer requirements when regridding
-            Array<int> tag_buffer;
-            setupTagBuffer(tag_buffer, d_scratch_gridding_algorithm);
-
-            // TODO: d_current_time is actually nan here. Since we don't do
-            // any sort of tagging based on the current time I think we can
-            // just put in a bogus (nonnegative) value.
-            d_scratch_gridding_algorithm->regridAllFinerLevels(
-                d_scratch_hierarchy, 0, 0.0 /*d_current_time*/, tag_buffer);
-            if (d_do_log) plog << "IBFEMethod: finished scratch hierarchy regrid" << std::endl;
-        }
-
-        // At this point in the code SAMRAI has already redistributed the
-        // patches (usually by taking into account the number of IB points on
-        // each patch). Here is the other half: if requested, we inform
-        // libMesh of the updated partitioning so that libMesh Elems and Nodes
-        // are on the same processor as the relevant SAMRAI patch.
-        if (d_libmesh_partitioner_type == SAMRAI_BOX)
-        {
-            for (unsigned int part = 0; part < d_num_parts; ++part)
-            {
-                EquationSystems& equation_systems = *d_active_fe_data_managers[part]->getEquationSystems();
-                MeshBase& mesh = equation_systems.get_mesh();
-                BoxPartitioner partitioner(*d_hierarchy, equation_systems.get_system(COORDS_SYSTEM_NAME));
-                partitioner.repartition(mesh);
-            }
-        }
+    // Set up the coupling matrix which will be used by each system.
+    d_diagonal_system_coupling.resize(NDIM);
+    for (unsigned int i = 0; i < NDIM; ++i)
+        for (unsigned int j = 0; j < NDIM; ++j) d_diagonal_system_coupling(i, j) = i == j ? 1 : 0;
 
+    // Create the FE data managers that manage mappings between the FE mesh
+    // parts and the Cartesian grid.
+    d_equation_systems.resize(d_meshes.size());
+    d_primary_fe_data_managers.resize(d_meshes.size(), nullptr);
+    d_scratch_fe_data_managers.resize(d_meshes.size(), nullptr);
+    d_active_fe_data_managers.resize(d_meshes.size(), nullptr);
+    d_fe_data.resize(d_meshes.size(), nullptr);
```

Comment:
we don't need the `nullptr`s

Reply:
👍 

I removed some others while I was at it...